gidgethub/abc.py

```diff
@@ -73,37 +95,62 @@ def __init__(self, requester: str, *, oauth_token: Opt[str] = None,
         return data, more
 
     async def getitem(self, url: str, url_vars: Dict = {},
-                      *, accept: str = sansio.accept_format()) -> Any:
+                      *, accept: str = sansio.accept_format(),
+                      auth_type: Opt[str] = None,
+                      token: Opt[str] = None
+                      ) -> Any:
         """Send a GET request for a single item to the specified endpoint."""
-        data, _ = await self._make_request("GET", url, url_vars, b"", accept)
+        data, _ = await self._make_request("GET", url, url_vars, b"", accept,
+                                           auth_type=auth_type, token=token)
         return data
 
     async def getiter(self, url: str, url_vars: Dict = {},
-                      *, accept: str = sansio.accept_format()) -> AsyncGenerator[Any, None]:
+                      *, accept: str = sansio.accept_format(),
+                      auth_type: Opt[str] = None,
+                      token: Opt[str] = None
+                      ) -> AsyncGenerator[Any, None]:
         """Return an async iterable for all the items at a specified endpoint."""
-        data, more = await self._make_request("GET", url, url_vars, b"", accept)
+        data, more = await self._make_request("GET", url, url_vars, b"", accept,
+                                           auth_type=auth_type, token=token)
         for item in data:
             yield item
         if more:
             # `yield from` is not supported in coroutines.
-            async for item in self.getiter(more, url_vars, accept=accept):
+            async for item in self.getiter(more, url_vars, accept=accept,
+                                           auth_type=auth_type, token=token):
                 yield item
 
     async def post(self, url: str, url_vars: Dict = {}, *, data: Any,
-                   accept: str = sansio.accept_format()) -> Any:
-        data, _ = await self._make_request("POST", url, url_vars, data, accept)
+                   accept: str = sansio.accept_format(),
+                   auth_type: Opt[str] = None,
```

Comment:
As far as I know, only one API accepts the `bearer jwt` authorization type, and it is a POST method.
I've added the same `auth_type` and `token` arguments to the other methods for symmetry, but now I'm not sure about it. Maybe only `post` should have these arguments.

Reply:
I had some more time to experiment with GitHub APIs. Basically everything in listed in this [documentation](https://developer.github.com/v3/apps/) can be accessed using json web token. It means `getitem`, `getiter` and `post` will benefit from the `auth_type` params.